exercises/concept/calculator-conundrum/.meta/src/reference/java/CalculatorConundrum.java

```diff
@@ -0,0 +1,33 @@
+public class CalculatorConundrum {
+
+    public String calculate(int operand1, int operand2, String operation) {
+        int result;
+        try {
+            if (operation == null) {
+                throw new IllegalArgumentException("Operation cannot be null");
+            }
+            switch (operation) {
+                case "+":
+                    result = SimpleOperation.addition(operand1, operand2);
+                    break;
+                case "*":
+                    result = SimpleOperation.multiplication(operand1, operand2);
+                    break;
+                case "/":
+                    result = SimpleOperation.division(operand1, operand2);
+                    break;
+                case "":
+                    throw new IllegalArgumentException("Operation cannot be empty");
+                default:
+                    throw new IllegalArgumentException(String.format("%s operation does not exist", operation));
+            }
+        } catch (ArithmeticException | IllegalArgumentException e){
+            if (e instanceof ArithmeticException) {
+                throw new ArithmeticException("Divide by zero operation illegal");
```

Comment:
This too seems like it is an illegal operatio, so IllegalOperationException makes perfect sense here.

Also, we should encourage best practices here so they should also be including the caught exception as the cause for this one.

It means we need to add a constructor to IllegalOperationException that takes a String message and Throwable cause. Since we extend IllegalArgumentException, we can call `super(message, cause)` for that constructor.